### DISCUSSION TITLE
Plotting throws Offset too large exceptions

### DISCUSSION THREAD

**Discussion by claudiumusat** [Chia-Network/chia-blockchain#3253] [Support]
Hello,

It's my first time plotting and on my main machine things are going haywire:

**Describe the bug**
On an 8 core machine with an external Sandisk 2TB SSD, plotting always results in an InvalidStateException:

Log:
2021-04-30T18:35:39.568  chia.plotting.create_plots       : [32mINFO    [0m Creating 1 plots of size 32, pool public key:  8b306efdc4eabdb7adc1251657e8fd05e35a7e7b54aa71c25f6a9c08f10f0a6e27359d3de80f8c059ee6510475171563 farmer public key: b2c74a8b2bdbc6080fdacb02ca3705ae32d3ff757ec89bc4da73a583c068d82b6565aa2b83acb2c131fb86630599864e[0m
2021-04-30T18:35:39.574  chia.plotting.create_plots       : [32mINFO    [0m Memo: 8b306efdc4eabdb7adc1251657e8fd05e35a7e7b54aa71c25f6a9c08f10f0a6e27359d3de80f8c059ee6510475171563b2c74a8b2bdbc6080fdacb02ca3705ae32d3ff757ec89bc4da73a583c068d82b6565aa2b83acb2c131fb86630599864e6c7f8a19e35ae3cae93332c6feb1c2b96cd8dfea6207685c85f628da4ec5bd78[0m
2021-04-30T18:35:39.575  chia.plotting.create_plots       : [32mINFO    [0m Starting plot 1/1[0m

Starting plotting progress into temporary dirs: /media/c/Extreme SSD/chia-temp3 and /media/c/Extreme SSD/chia-temp3
ID: fe60cea616db31b574697341172dd1983a6a9e2735c10581df7e6038e9b2d6d2
Plot size is: 32
Buffer size is: 3390MiB
Using 128 buckets
Using 2 threads of stripe size 65536

Starting phase 1/4: Forward Propagation into tmp files... Fri Apr 30 18:35:39 2021
Computing table 1
F1 complete, time: 123.079 seconds. CPU (141.71%) Fri Apr 30 18:37:42 2021

Computing table 2
	Bucket 0 uniform sort. Ram: 3.251GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.

....

	Bucket 124 uniform sort. Ram: 3.251GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 125 uniform sort. Ram: 3.251GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 126 uniform sort. Ram: 3.251GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 127 uniform sort. Ram: 3.251GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Total matches: 4294983014
Forward propagation table time: 1474.752 seconds. CPU (141.820%) Fri Apr 30 19:02:17 2021
Computing table 3
	Bucket 0 uniform sort. Ram: 3.251GiB, u_sort min: 2.250GiB, qs min: 0.563GiB.
terminate called after throwing an instance of 'InvalidStateException'
  what():  Offset too large: 41894

**To Reproduce**
it is not always reproducible - sometimes plotting stops with no error logged.

**Expected behavior**
Plotting stops at 6%

**Desktop (please complete the following information):**

- OS: Linux
- OS Version/Flavor: Ubuntu 20.4 64bit
- CPU: Intel i7-7700K

**Note: Plotting on the same machine on the internal HDD works flawlessly.**

Thanks for your help!

**Comment by razamataz** [comment]
do not use EXTERNAL disks for plotting

**Comment by sargonas** [comment]
This is likely due to the sleep state or other power saving functionality stemming from using an external drive. It's highly ill advised for plotting as a combination of the latency of going over USB controllers, power saving features of the devices, and more are going to compound a lot of bad news into your reliability.

**Comment by NessDan** [comment]
Just wanted to add to this - I'm seeing this happen with my setup. I'm using 2 [Asus HYPER M.2 X16](https://www.asus.com/us/Motherboards-Components/Motherboards/Accessories/HYPER-M-2-X16-CARD/) cards. One has 4x 2TB NVMes and the other has 3x.

I'll be playing with the BIOS settings to see if I can figure out why this error is happening...

**Comment by room101-dev** [comment]
Sounds like You have already DESTROYED your SSD

You can only plot with NVME drives m.2, and you can use the SSD for temp-storage (-2)

If u plot a normal sata type ssd ( samsung evo 870 or wd-blue ) you will kill it in one day, and still not get a 'plot', if you plot on an HDD you will murder the HDD forever.

Why in the hell do people do this? I suspect its because 'CHIA is Green", and you can use old hardware, wake up dude the chia-net dev owners are professional liars.

...

In my past 3+ months playing with chia, I have killed two HDD's and two NVME's, I haven't killed a ssd-sata cuz I knew better, the first hdd I killed was just for fun, I wanted to see what 'plot create' would do, and sure enough after 8hrs of grinding the drive was complete trashed. My second hdd loss was from the 'chia plots check', what they call test&verify, and it MURDERED my hdd storage drive causing me to lose 51 plots.  Then I killed two NVME's early on, cuz I didn't know that the 'plots create' software didn't have a throtte, so I patched the code so that now it idles if the nvme goes over 70C, an also now I keep fans on all my NVME's I have 3 on each mache.

I hope u and the guy above who wants to 'plot' on a mobile phone are beginning to understand that all of chia is 'BS myth of Feces', its not green, and if you use old&slow hw, it will die, and be destroyed.

**Comment by BrandtH22** [comment]
@claudiumusat , since this ticket was opened a number of years ago without further comments it seems this issue has been resolved. 
We will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions or comments.

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
